plugins/catalog-react/src/components/EntityOwnerPicker/useFacetsEntities.ts

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { useApi } from '@backstage/core-plugin-api';
+import useAsyncFn from 'react-use/lib/useAsyncFn';
+import { catalogApiRef } from '../../api';
+import { useState } from 'react';
+import { Entity, parseEntityRef } from '@backstage/catalog-model';
+
+type FacetsCursor = {
+  start: number;
+  text: string;
+};
+
+type FacetsEntitiesResponse = {
+  items: Entity[];
+  cursor?: string;
+};
+
+type FacetsInitialRequest = {
+  text: string;
+};
+
+export function useFacetsEntities({ enabled }: { enabled: boolean }) {
```

Comment:
I think a commend describing these internal hooks could help future you and others 😁